src/controllers/user.controller.ts

```diff
@@ -174,13 +221,61 @@ export const acceptGroupInvitation = async (req: Request, res: Response) => {
   }
 };
 
-export const getFriends = async (req: Request, res: Response) => {
-  const userId = req.params?.userId!.toString();
+/**
+ * An async function that enables a user to leave a group they are currently in
+ * @param {Request} req - Express request object.
+ * @param {Response} res - Express response object.
+ * @returns {Response} - A JSON response indicating success or failure or error message with a 500 status code.
+ */
+export const leaveGroup = async (req: Request, res: Response) => {
+  const userId = req.params?.userId || '';
+  const groupId = req.query?.groupId?.toString() || '';
```

Comment:
do we need to pull a `groupId` in this controller? A user can only be in one group at a time, so can't we just reset the user's `currentGroup` field (if it is defined)?

Reply:
we should also consider adding logic to delete the group if you are the last member in it.